### ISSUE TITLE
Support for union type

### ISSUE THREAD

**Issue by guyariely** [anymaniax/orval#551]
The following schema:

```yaml
Example:
      type: object
      properties:
        prop:
          type:
            - string
            - number
```

Will yield the following type:

```yaml
interface Example {
  prop: unknown;
}
```

While the appropriate type would be:

```yaml
interface Example {
  prop: string | number;
} 
```

It seems a property with more than one type is being treated as `unknown`.
This is especially apparent when trying to document nullable properties.

**Package Version:** 6.9.6

**Comment by mb21** [comment]
That would be great, also when using OpenAPI's `oneOf`.

Would be like [`openapi-typescript-codegen`'s `--useUnionTypes` option](https://github.com/ferdikoomen/openapi-typescript-codegen/blob/master/docs/enum-vs-union-types.md).

**Comment by DominicBach** [comment]
FYI this is not a valid OpenAPI specification, see https://swagger.io/specification/#schema-object
> The following properties are taken from the JSON Schema definition but their definitions were adjusted to the OpenAPI Specification.
>
> type - Value MUST be a string. Multiple types via an array are not supported.

The correct way to model this would be
```yaml
Example:
  type: object
  oneOf:
    - type: object
      properties:
        prop:
          type: string
    - type: object
      properties:
        prop:
          type: number
```

**Comment by mb21** [comment]
@DominicBach I updated the snipped to be complete (now it validates) and replaced the confusing double use of `type` with `kind` in one place. Please check out the links, hope it makes sense now?

**Comment by geirsagberg** [comment]
@DominicBach Type arrays are supported in OpenAPI 3.1: https://www.openapis.org/blog/2021/02/16/migrating-from-openapi-3-0-to-3-1-0

**Comment by geirsagberg** [comment]
I added a PR for nullability that should fix this one as well: #645
